sources/functions/qbittorrent

```diff
@@ -41,48 +36,110 @@ function whiptail_qbittorrent() {
+            if dpkg -s libtorrent-rasterbar | grep -q ${stdcver} && whiptail_skip_libtorrent_rasterbar; then
+                return
+            fi
+        fi
+    fi
+    false
+    return
+}
+
+function build_libtorrent_qbittorrent() {
+    booststrap
+    echo "using gcc : : : <cflags>\"-march=native -std=${stdcver}\" <cxxflags>\"-march=native -std=${stdcver}\" ;" > /root/user-config.jam
+
+    cd /tmp
+    git clone -b ${libtorrent_branch} https://github.com/arvidn/libtorrent >> ${log} 2>&1
+    cd /tmp/libtorrent
+    if [[ -f /root/libtorrent-${libtorrent_branch}.patch ]]; then
+        patch -p1 < /root/libtorrent-${libtorrent_branch}.patch >> ${log} 2>&1
+    fi
+    VERSION=$(grep AC_INIT configure.ac | grep -oP '\d+\.\d+\.\d+')
+    /opt/boost_${BOOST_VERSION}/b2 -j$(nproc) crypto=${cryptography_type} variant=release link=static install --prefix=/tmp/dist/libtorrent/usr/local install >> ${log} 2>&1
+    tag=" static with ${stdcver} march=native"
+    mkdir -p /root/dist
+    fpm -f -C /tmp/dist/libtorrent -p /root/dist/libtorrent-rasterbar_VERSION.deb -s dir -t deb -n libtorrent-rasterbar --version ${VERSION} --description "Libtorrent rasterbar compiled by swizzin$tag" >> ${log} 2>&1
+    rm_if_exists /tmp/libtorrent
+    rm_if_exists /tmp/dist
+    dpkg -i /root/dist/libtorrent-rasterbar_${VERSION}.deb >> ${log} 2>&1 || {
+        echo_error "Libtorrent could not be installed. Setup will not proceed"
+        exit 1
+    }
+}
+
 function build_qbittorrent() {
     if [[ $qbittorrent == repo ]]; then
         apt_install qbittorrent-nox
     else
-        if ! check_ram 2G; then
-            swap=true
-            tmp_swap_on
-        fi
         noexec=$(grep "/tmp" /etc/fstab | grep noexec)
         if [[ -n $noexec ]]; then
             mount -o remount,exec /tmp
             noexec=1
         fi
         VERSION=${qbittorrent}
-        if [[ $(lsb_release -cs) == focal && $deluge == 1.3-stable ]]; then
-            LIST='build-essential pkg-config automake libtool git zlib1g-dev libssl-dev libgeoip-dev libboost1.67-dev libboost-system1.67-dev libboost-chrono1.67-dev libboost-random1.67-dev python3 qtbase5-dev qttools5-dev-tools zlib1g-dev'
-        else
-            LIST='build-essential pkg-config automake libtool git zlib1g-dev libssl-dev libgeoip-dev libboost-dev libboost-system-dev libboost-chrono-dev libboost-random-dev python3 qtbase5-dev qttools5-dev-tools zlib1g-dev'
-        fi
+        LIST='build-essential automake pkg-config libtool git zlib1g-dev libssl-dev libgeoip-dev python3 qtbase5-dev qttools5-dev-tools zlib1g-dev'
+
         apt_install --no-recommends $LIST
         cd /tmp
         rm_if_exists "/tmp/qbittorrent"
-        wget -q https://github.com/qbittorrent/qBittorrent/archive/release-${VERSION}.tar.gz >> "${log}" 2>&1
+        wget https://github.com/qbittorrent/qBittorrent/archive/release-${VERSION}.tar.gz >> "${log}" 2>&1
```

Comment:
Please let's make these full path and not pwd relative while this is being changed

Reply:
@liaralabs not sure why this was marked as resolved, you can get rid of the `cd /tmp`. If you need to be in the same dir for `./configure`, might as well ` cd /tmp/qbitttorrent` just before it. None of the paths between L121 and L 126 need to be based on cwd from what I can see.

Reply:
I believe you are commenting on outdated code here again.

Reply:
Yup, looks like it. And it looks like you already thought of the exact same solution i had in mind ha!